site2/docs/concepts-messaging.md

```diff
@@ -55,6 +55,8 @@ If batching is enabled, the producer will accumulate and send a batch of message
 
 A consumer is a process that attaches to a topic via a subscription and then receives messages.
 
+A consumer pulls messages from a broker. There is a queue at the consumer side to receive messages pushed from the broker. The queue size is configurable by [`receiverQueueSize`](client-libraries-java.md#configure-consumer) (default: 1000). Each time `consumer.receive()` is called, a message is dequeued from the buffer. If the requested message size of a consumer is larger than the queue size, after all messages in the queue are consumed, messages are delivered to the consumer immediately once there comes new messages in the queue.  
```

Comment:
> If the requested message size of a consumer is larger than the queue size, after all messages in the queue are consumed, messages are delivered to the consumer immediately once there comes new messages in the queue.  

Consumer will pull more messages while available messages in the receive queue < half of the queue size.